enterprise/views.py

```diff
@@ -972,3 +990,271 @@ def get(self, request, enterprise_uuid, course_id):
+        'expected_learning_items_header': _("What you'll learn"),
+        'view_expected_learning_items_text': _('See More'),
+        'hide_expected_learning_items_text': _('See Less'),
+        'confirm_button_text': _('Confirm Program'),
+        'summary_header': _('Program Summary'),
+        'price_text': _('Price'),
+        'length_text': _('Length'),
+        'length_info_text': _('{}-{} weeks per course'),
+        'effort_text': _('Effort'),
+        'effort_info_text': _('{}-{} hours per week, per course'),
+        'program_not_eligible_for_one_click_purchase_text': _('Program not eligible for one-click purchase.'),
+    }
+
+    def get_program_details(self, request, program_uuid):
+        """
+        Retrieve fundamental details used by both POST and GET versions of this view.
+
+        Specifically:
+
+        * Take the program UUID and get specific details about the program.
+        * Determine whether the learner is enrolled in the program.
+        """
+        try:
+            program_details = CourseCatalogApiServiceClient().get_program_by_uuid(program_uuid)
+        except (HttpClientError, ImproperlyConfigured) as error:
+            LOGGER.error('Failed to get program details for program with UUID %s due to: %s', program_uuid, str(error))
+            raise Http404
+
+        if program_details is None:
+            LOGGER.error('No program details returned for program with UUID %s', program_uuid)
+            raise Http404
+
+        # Extend our program details with context we'll need for display or for deciding redirects.
+        program_details = ProgramDataExtender(program_details, request.user).extend()
+
+        # TODO: Upstream this additional context to the platform's `ProgramDataExtender` so we can avoid this here.
+        program_details['enrolled_in_program'] = False
+        enrollment_count = 0
+        for course in program_details['courses']:
+            course_run = course['course_runs'][0]
+            if course_run['is_enrolled'] and course_run['upgrade_url'] is None:
+                # We're enrolled in the program if we have certificate-eligible enrollment in even 1 of its courses.
+                program_details['enrolled_in_program'] = True
+                enrollment_count += 1
+
+        # We're certificate eligible for the program if we have certificate-eligible enrollment in all of its courses.
+        program_details['certificate_eligible_for_program'] = (
+            enrollment_count == len(program_details['courses'])
+        )
+        return program_details
+
+    def get_enterprise_program_enrollment_page(self, request, enterprise_customer, program_details):
+        """
+        Render Enterprise-specific program enrollment page.
+        """
+        # Safely make the assumption that we can use the first authoring organization.
+        organization = program_details['authoring_organizations'][0]
```

Comment:
As per the comment, I was told from @douglashall that we can assume the first is fine to use. We do something similar in the course enrollment view with `organization = course['owners'][0]`. Either way, I'll cover this case and return `{}` otherwise, and `.get()` things from it below..